src/coreclr/jit/importer.cpp

```diff
@@ -8685,6 +8687,24 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                                   combine(combine(CORINFO_CALLINFO_ALLOWINSTPARAM, CORINFO_CALLINFO_SECURITYCHECKS),
                                           (opcode == CEE_CALLVIRT) ? CORINFO_CALLINFO_CALLVIRT : CORINFO_CALLINFO_NONE),
                                   &callInfo);
+
+                    if (callInfo.kind == CORINFO_CALL &&
```

Comment:
Does this need to be skipped for debug code?

Reply:
`impTryFindField` calls `canInline` which does this check
https://github.com/dotnet/runtime/blob/9fa5128e2d16855bef83560a5a38b493fa3eb7ea/src/coreclr/vm/jitinterface.cpp#L7879-L7887

Reply:
This check is for callee. Inlining needs to be disabled for debuggable caller too. It is typically done by checking `opts.compDbgCode` in the JIT.